src/Microsoft.AspNetCore.Razor.Language/Intermediate/TypeParameter.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Language.Intermediate
+{
+    public sealed class TypeParameter
```

Comment:
Do we plan to add more properties to this later? Curious why this is its own type.

Reply:
Yes, if we add constraints or variance annotations we would add them to this type. I originally started with as string and updated it after I sent the PR